modules/gdscript/gdscript_parser.h

```diff
@@ -918,6 +918,7 @@ class GDScriptParser {
 		Vector<IdentifierNode *> captures;
 		HashMap<StringName, int> captures_indices;
 		bool use_self = false;
+		bool is_in_static_context = false; // For analyzer. Just for don't modify `FunctionNode::is_static`.
```

Comment:
I'm not sure about this, but `FunctionNode::is_static` is more of a parser flag that indicates whether there is `static` before `func`.

Reply:
I don't think it would be a big deal to use the `FunctionNode::is_static` for this, as the lambdas cannot have the `static` keyword anyway, but it's probably clearer to have a dedicated flag for it.

Reply:
There's a grammar issue in the comment, do you mean this?
```suggestion
		bool is_in_static_context = false; // For analyzer, to avoid modifying `FunctionNode::is_static`.
```

Reply:
That's not exactly what I meant. This is for distinguish between `static func` syntax (which is not allowed for lambdas) and a flag that a lambda is in static context (so it cannot use non-static things).

```suggestion
		bool is_in_static_context = false; // Dedicated flag for lambdas, dont't confuse with `FunctionNode::is_static`.
```

I'm still not sure if we should differentiate this, maybe one flag is better. We can simply add a comment that the static flag of regular functions is determined by the explicit `static` keyword in the parser, and the static flag of lambdas is determined implicitly, from the context (you can't specify it explicitly).